service/controller/app/v1/resource/kubeconfigfinalizer/create.go

```diff
@@ -0,0 +1,67 @@
+package kubeconfigfinalizer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if key.InCluster(cr) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("app %#q does not use a kubeconfig secret for apps in the control plane", cr.GetName()))
```

Comment:
I would have had a more generic message here, mentioning control plane here is a bit confusing.
I understand that app using InCluster are installed on the control-plane, but I feel like this bit of information is too much here.

```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("app %#q does not use a kubeconfig secret", cr.GetName()))
```

Reply:
Agreed, maybe it's too much information 👍